tap-version-14-specification.md

```diff
@@ -0,0 +1,1167 @@
+   delimiting a directive, provided the `\` was not escaped.
+3. Treat any unescaped `#` as a literal `#` in a description or reason
+   field, if doing so would not cause it to be treated as a delimiter.
+
+### Escaping Examples
+
+The following are examples of escaping `#` and `\` characters, and
+including unescaped `#` characters in the Test Point description or `TODO`
+reason.
+
+Note that the specific fields `description`, `todo`, and `todo reason` are
+not normative, and shown for illustration purposes only.  Harnesses
+_should_ present this data to their consumers in whatever manner is
+appropriate for their language and context.
+
+```tap
+TAP version 14
+
+# description: hello
+# todo: true
+ok 1 - hello # todo
+
+# description: hello # todo
+# todo: false
+ok 2 - hello \# todo
+
+# description: hello
+# todo: true
+# todo reason: hash # character
+ok 3 - hello # todo hash \# character
+# (assuming "character" isn't a known custom directive)
+ok 4 - hello # todo hash # character
+
+# description: hello \
+# todo: true
+# todo reason: hash # character
+ok 5 - hello \\# todo hash \# character
+# (assuming "character" isn't a known custom directive)
+ok 6 - hello \\# todo hash # character
+
+# description: hello # description # todo
+# todo: false
+# (assuming "description" isn't a known custom directive)
+ok 7 - hello # description # todo
+
+# multiple escaped \ can appear in a row
+# description: hello \\\# todo
+# todo: false
+ok 8 - hello \\\\\\\# todo
+
+1..8
+```
+
+## Subtests
+
+Subtests provide a way to nest one TAP14 stream inside another.  This is
+useful in a variety of situations.  For example:
+
+1. A Harness parses a collection of TAP documents, providing output which
+   is also in TAP format.
+
+    ```tap
+    TAP version 14
+    1..2
+
+    # Subtest: foo.tap
+        1..2
+        ok 1
+        ok 2 - this passed
+    ok 1 - foo.tap
+
+    # Subtest: bar.tap
+        ok 1 - object should be a Bar
+        not ok 2 - object.isBar should return true
+          ---
+          found: false
+          wanted: true
+          at:
+            file: test/bar.ts
+            line: 43
+            column: 8
+          ...
+        ok 3 - object can bar bears # TODO
+        1..3
+    not ok 2 - bar.tap
+      ---
+      fail: 1
+      todo: 1
+      ...
+    ```
+
+2. A test framework Producer provides an API for grouping assertions about
+   a related subject.  For example:
+
+    ```js
+    import t from 'tap'
+    t.ok(true, 'true is ok')
+    t.test('this is a subtest', subtest => {
+      subtest.pass('this is fine')
+      subtest.fail('this is not fine')
+      subtest.end()
+    })
+    ```
+
+    which produces the output:
+
+    ```tap
+    TAP version 14
+    ok 1 - true is ok
+    # Subtest: this is a subtest
+        ok 1 - this is fine
+        not ok 2 - this is not fine
+        1..2
+    not ok 2 - this is a subtest
+    1..2
+    ```
+
+Subtests are designed with graceful fallback for TAP 13 harnesses in mind.
```

Comment:
I think we are using `TAP14` for referring to the TAP 14 specification. In some parts of this doc we have `TAP13`, but also `TAP 13`. I think we need one last pass over the doc to use a single format, as with TAP14.